File: string_problems/DecodeSlantedCipherText.js

// problem link - https://leetcode.com/problems/decode-the-slanted-ciphertext/description/

let decodeCiphertext = function(encodedText, rows) {
    let len = encodedText.length;

    // step 1 - find number of column
    let col = Math.floor(len / rows);

    // step 2 - iterate on encoded string diagonally
    let originalText = "";

    for (let c = 0; c < col; c++) {
        for (let j = c; j < len; j += (col + 1)) {
            // step 3 - prepare original string
            originalText += encodedText[j];
        }
    }

    // step 4 - remove trailing spaces
    while (originalText.length > 0 && originalText[originalText.length - 1] === ' ') {
        originalText = originalText.slice(0, -1);
    }

    // step 5 - return result
    return originalText;
};

File: string_problems/DecodeSlantedCiphertext.cpp

// Problem link - https://leetcode.com/problems/decode-the-slanted-ciphertext/description/?envType=daily-question&envId=2026-04-04

#include<bits/stdc++.h>
using namespace std;

string decodeCiphertext(string encodedText, int rows) {
    int len =encodedText.length();

    // step 1 - find number of column
    int col = len / rows;

    // step 2 - iterate on encoded string diagonally
    string originalText = "";
    for(int c=0; c<col; c++){
        for(int j=c; j<len; j+= (col+1)){
            // step 3 - prepare original string
            originalText += encodedText[j];
        }
    }

    // step 4 - extract trailing spaces
    while (!originalText.empty() && originalText.back() == ' '){
        originalText.pop_back();
    }

    // step 5 - return original string
    return originalText;
}

int main(){
    // string encodedText = "ch   ie   pr";
    // int rows = 3;

    // string encodedText = "iveo    eed   l te   olc";
    // int rows = 4;

    string encodedText = "coding";
    int rows = 1;
    cout<<decodeCiphertext(encodedText, rows);
    return 0;
}